### PR TITLE
CLOUD RFD-0029: Support running against teleport *and* cloud repos

### PR DESCRIPTION
**Pull request by gozer**
Should be a no-op for gravitational/teleport.

Just flips logic around for some review logic when in 'cloud' repo vs. 'teleport'

### CODE REVIEW

**Review by jimbishopp — COMMENTED** [review]
Minor refactor comments that should simplify some of the logic.

**Comment by gozer** [comment]
Thanks, all address all these shortly, stay tuned.

File: bot/internal/review/review.go [hunk 1]
```diff
@@ -131,8 +132,9 @@ func (r *Assignments) IsInternal(author string) bool {
 }
 
 // Get will return a list of code reviewers for a given author.
-func (r *Assignments) Get(author string, docs bool, code bool, files []github.PullRequestFile) []string {
+func (r *Assignments) Get(e env.Environment, docs bool, code bool, files []github.PullRequestFile) []string {
 	var reviewers []string
+	author := e.Author
```
[the remark this answers is not in the document]
Comment (gozer) [reply bot/internal/review/review.go:137]: Yup, I was indeed trying to keep the diff as small as possible to ease in reviewing/backporting changes.

**Review by jimbishopp — COMMENTED** [review]
Good to go with a couple tweaks while we're here (missed it earlier). I created a [branch](https://github.com/gravitational/shared-workflows/compare/main...jim/bot) with two small commits to consider.

- [Pass Values (except for large structs)](https://github.com/gravitational/shared-workflows/commit/8c9e20d7d40719c8aae3dc132339caab35d3e47f)
- [Remove unnecessary variable](https://github.com/gravitational/shared-workflows/commit/0618fc042b3f28301e3af8a558b7e17ab6981dd9)

**Comment by gozer** [comment]
> Good to go with a couple tweaks while we're here (missed it earlier). I created a [branch](https://github.com/gravitational/shared-workflows/compare/main...jim/bot) with two small commits to consider.
> 

Good tweaks, I'll merge them in. As you correctly pointed out earlier, until this was all done, I was attempting to keep the actual changes to the code as minimal as possible.

And how that this is the canonical source for this bot, no reasons not to improve.

We'll need to keep an eye on Teleport and make sure we don't cause unintended side effects there, but not overly worried.

Thanks!

**Comment by jimbishopp** [comment]
> Good tweaks, I'll merge them in. As you correctly pointed out earlier, until this was all done, I was attempting to keep the actual changes to the code as minimal as possible.

Yeah, I was concerned about this as well last night. But the changes make the PR diff even more minimal since Environment is loaded and passed around as a pointer elsewhere. We're now following the same pattern rather than dereferencing the pointer only in the review action.

**Comment by gozer** [comment]
@jimbishopp changes cherry-picked to this PR.

File: bot/internal/review/review.go [hunk 2]
```diff
@@ -237,23 +239,29 @@ func (r *Assignments) getAdminReviewers(author string) []string {
...
 		return reviewers[:n], reviewers[n:]
 	}
 
-	// Cloud gets reviewers assigned from Core.
 	team := v.Team
-	if v.Team == "Cloud" {
-		team = "Core"
+
+	// Cloud gets reviewers assigned from Core in Teleport
+	// Core gets reviewers assigned from Cloud in Cloud
```
Comment (jimbishopp) [inline bot/internal/review/review.go:256]: I thought I updated this to match the same comment below in `checkCodeReviews` but apparently not - late night coding...

```suggestion
	// Teams do their own internal reviews
```

**Review by jimbishopp — APPROVED** [review]

**Review by jimbishopp — APPROVED** [review]

**Pull request merged**